pkg/virt-handler/vm_test.go

```diff
@@ -801,6 +801,37 @@ var _ = Describe("VirtualMachineInstance", func() {
 			Expect(blockMigrate).To(BeTrue())
 			Expect(err).To(Equal(fmt.Errorf("cannot migrate VMI with non-shared PVCs")))
 		})
+		It("should fail migration for non-shared data volume PVCs", func() {
```

Comment:
it would worth adding a unit test that ensures we allow migrations for datavolumes that are readwritemany.